libs/dmlf/include/dmlf/execution/execution_params.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "variant/variant.hpp"
+#include <vector>
+
+using ExecutionParams = std::vector<fetch::variant::Variant>;
```

Comment:
I would rather alias `Variant` than `vector<Variant>`, as the user will have to handle `Variants` anyway 

```cpp
using Parameter = fetch::variant::Variant;
using ExecutionParams = std::vector<Parameter>
```
user code can than do 

```cpp
uto p3 = client->client_->Run(SERVER_PUB, "exe1", "state1", "add", {Parameter{"hi"}, Parameter{2}});
```

Reply:
I know what you mean but my thinking was that the function we're calling takes a list of parameters, so the "top level" vector represents the actual parameter list (some of which may then themselves be vectors).

Reply:
Oh, no, hang on, I think I've misunderstood you.

Reply:
Yes, I have, I see what you mean. Yes.

Reply:
sorry, I was not very clear in my comment.
Yes, I meant that `Run` method doesn't need to abstract away the fact that parameters are a list/vector of elements. The elements could be named Variant, Parameter, Variable or something, it doesn't matter as long as we provide a practical (generic, flexible) interface to set and get values of different types out of them.

Reply:
Done!